### ISSUE TITLE
VControl does not use SvgDocument to computeSize

### ISSUE THREAD

**Issue by edward-k-francis** [EclipseNebula/nebula#250]
If a VControl only contains a SvgDocument the size is only 2x2 (plus margins) instead of using the width/height or viewBox of the SvgDocument.

**Comment by lcaron** [comment]
Hi,

Can you please provide a snippet with a sample svg file ?

Thank you

Le jeu. 20 août 2020 à 06:56, edward-k-francis <notifications@github.com> a
écrit :

**Comment by edward-k-francis** [comment]
@lcaron, here is a snippet of code where the computeSize of the canvas is too small for the Svg image.
[caretLeft.zip](https://github.com/eclipse/nebula/files/5104903/caretLeft.zip)



```
SvgDocument SVG_BACK = null;
try (final InputStream in = SVGCanvasExample.class.getResourceAsStream("icons/caretLeft.svg")) {
     SVG_BACK = SvgDocument.load(in);
}
catch (IOException e) {
    e.printStackTrace();
}
if(SVG_BACK == null) {
    return;
}

VCanvas vCanvas = new VCanvas(this,SWT.NONE);
        
VPanel vPanel = vCanvas.getPanel();

VGridLayout layout = new VGridLayout();
layout.marginHeight = 0;
layout.marginWidth = 0;
vPanel.setLayout(layout);

VLabel vLabel = new VLabel(vPanel, SWT.NONE);
vLabel.setImage(SVG_BACK);
vLabel.setBackground(Display.getDefault().getSystemColor(SWT.COLOR_BLACK));

Point size = vCanvas.computeSize(SWT.DEFAULT, SWT.DEFAULT);
```

**Comment by lcaron** [comment]
Hi,

Thank you for giving this information, I'll have a look !

Laurent

Le jeu. 20 août 2020 à 20:48, edward-k-francis <notifications@github.com> a
écrit :

**Comment by lcaron** [comment]
I've also created a bug in Bugzilla : https://bugs.eclipse.org/bugs/show_bug.cgi?id=566500

**Comment by wimjongman** [comment]
@edward-k-francis Edward, can you also supply a patch for this?

**Comment by lcaron** [comment]
Bug fixed in Nebula 2.5.0
